terraform/routines.tf

```
@@ -1,6 +1,6 @@
 resource "google_bigquery_routine" "empty_to_null" {
   dataset_id      = local.routines_dataset
-  definition_body = "IF(LOWER(TRIM(x)) IN ('none', 'null', ''), NULL, x)"
+  definition_body = "IF(LOWER(TRIM(x)) IN ('none', 'null', 'nan', 'nat', ''), NULL, x)"
```

```suggestion
  definition_body = "IF(LOWER(TRIM(SAFE_CAST(x AS STRING))) IN ('none', 'null', 'nan', 'nat', ''), NULL, x)"
```
Given that your function only works on string values, you might as well cast the input to handle the cases where the input is not a string

---

Or you could even use normal `CAST` as a sort of type check